users/models.py

```diff
@@ -40,6 +52,10 @@ class User(AbstractBaseUser, PermissionsMixin, TimeStamp):
     USERNAME_FIELD = "email"
     REQUIRED_FIELDS = ["name"]
 
+    # pylint: disable=no-member
+    role = models.ForeignKey(Role, on_delete=models.CASCADE, default=1)
```

Comment:
I don't think deleting a user if a Role is deleted would be a good idea so in this case, we should not use `CASCADE`. instead, if a Role is getting deleted we should let the default value set in its place. 

When we follow the above approach we should make sure that one of the role always remain in the Roles table, Let's call it `System Admin` role and make sure it's not deleted ever.

Reply:
I have changed `models.CASCADE` to  `models.SET_DEFAULT`. Also for `System Admin` i have overrided the `delete` method of models and raise an `OperationError` if the role being `deleted` is `System Admin`. On a side note, wouldn't it be a better approach to use `models.PROTECT` to ensure that a `Role` can not be deleted if a `User` has that role. This way, new role would be created and Users would be assigned that new role and then finally the old role can be removed

Reply:
I agree with your proposed approach of protecting.